tests/unit/s2n_hkdf_test.c

```diff
@@ -419,19 +415,38 @@ int main(int argc, char **argv)
     for (uint8_t i = 0; i < NUM_TESTS; i++) {
         struct hkdf_test_vector *test = &tests[i];
 
-        s2n_blob_init(&in_key_blob, test->in_key, test->in_key_len);
-        s2n_blob_init(&salt_blob, test->salt, test->salt_len);
-        s2n_blob_init(&info_blob, test->info, test->info_len);
-        s2n_blob_init(&actual_prk_blob, test->pseudo_rand_key, test->prk_len);
-        s2n_blob_init(&actual_output_blob, test->output, test->output_len);
+        EXPECT_SUCCESS(s2n_blob_init(&in_key_blob, test->in_key, test->in_key_len));
+        EXPECT_SUCCESS(s2n_blob_init(&salt_blob, test->salt, test->salt_len));
+        EXPECT_SUCCESS(s2n_blob_init(&info_blob, test->info, test->info_len));
+        EXPECT_SUCCESS(s2n_blob_init(&actual_prk_blob, test->pseudo_rand_key, test->prk_len));
+        EXPECT_SUCCESS(s2n_blob_init(&actual_output_blob, test->output, test->output_len));
+        EXPECT_SUCCESS(s2n_blob_init(&prk_result, prk_pad, sizeof(prk_pad)));
+        EXPECT_SUCCESS(s2n_blob_init(&out_result, output_pad, sizeof(output_pad)));
 
         EXPECT_SUCCESS(s2n_hkdf_extract(&hmac, test->alg, &salt_blob, &in_key_blob, &prk_result));
         EXPECT_EQUAL(memcmp(prk_pad, actual_prk_blob.data, actual_prk_blob.size), 0);
 
+        /* The size of the PRK output should match the digest size */
+        uint8_t digest_size = 0;
+        EXPECT_SUCCESS(s2n_hmac_digest_size(test->alg, &digest_size));
+        EXPECT_EQUAL(prk_result.size, digest_size);
+
         EXPECT_SUCCESS(s2n_hkdf(&hmac, test->alg, &salt_blob, &in_key_blob, &info_blob, &out_result));
         EXPECT_EQUAL(memcmp(output_pad, actual_output_blob.data, actual_output_blob.size), 0);
     }
 
+    /* Ensure that the PRK output size can't be set beyond its max size */
+    {
+        s2n_stack_blob(small_prk_output, 10, 10);
+        EXPECT_FAILURE_WITH_ERRNO(s2n_hkdf_extract(&hmac, S2N_HMAC_SHA512, &salt_blob, &in_key_blob, &small_prk_output),
+                S2N_ERR_SAFETY);
```

Comment:
S2N_ERR_SAFETY is pretty general, so you might want this test to include a "sanity check" that if the output is larger than it doesn't fail.